### PR TITLE
Features/bugfix for conn

### PR DESCRIPTION
**Pull request by RyanKung**

### CODE REVIEW

**Comment by RyanKung** [comment]
1. Fixed `IceServer` list.
2. Support mDNS query and gather.
3. Support setting external_ip address.
4. Handshake info should always contain candidate info.

File: bin/node.rs [hunk 1]
```diff
@@ -222,7 +226,16 @@ async fn daemon_run(
     )?;
     let sig = key.sign(&auth.to_string()?).to_vec();
     let session = SessionManager::new(&sig, &auth, &temp_key);
-    let swarm = Arc::new(Swarm::new(stuns, key.address(), session.clone()));
+    let swarm = if let Some(eip) = external_ip {
```
Comment (googolmo) [inline bin/node.rs:229]: Maybe here can call swarm::new_with_external_ip(...,external_ip) directly.

**Review by Ma233 — APPROVED** [review]
LGTM

File: bin/node.rs (hunk 1 above)
Comment (Ma233) [reply bin/node.rs:237]: We may unify external interface of core mod in another pr. Then we can add a `SwarmBuilder` for the `Swarm` interface.

**Pull request merged**